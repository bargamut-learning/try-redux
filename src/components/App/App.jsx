import * as React from 'react';
import PostsList from '../PostsList/PostsListContainer';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.createPost = props.createPost;

    this.postTitleInput = React.createRef();
  }

  handleCreatePost = (event) => {
    event.preventDefault();

    this.createPost({ title: this.postTitleInput.current.value });
  };

  render() {
    const { text = `Default title`, image } = this.props;

    return (
      <>
        <h1>{this.text}</h1>

        <form onSubmit={this.handleCreatePost}>
          <input ref={this.postTitleInput} />

          <input type="submit" value="Create post" />
        </form>

        <PostsList />

        <img src={image} />
      </>
    );
  }
}

export default App;
