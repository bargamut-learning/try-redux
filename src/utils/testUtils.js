'use strict';

import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from '../store/reducers/rootReducer';

function render(
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = {}
) {
  const Wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
