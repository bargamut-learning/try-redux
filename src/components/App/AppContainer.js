import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import createPost from '../../store/actionCreators/createPost';

import App from './App';

const mapStateToProps = (state, ownProps) => ({
  posts: state.posts,
});

const mapDispatchToProps = {
  createPost,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
