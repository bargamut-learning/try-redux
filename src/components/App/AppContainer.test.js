'use strict';

import * as React from 'react';
import { render, screen, fireEvent } from '../../utils/testUtils';

import App from './AppContainer';

const mock = {
  text: `Test data`,
};

it('should render App correctly without data', () => {
  const { container } = render(<App />);

  expect(container).toMatchSnapshot();
});

it('should render App correctly with mock title', () => {
  const { text: title } = mock;

  const { container } = render(<App text={title} />, {
    initialState: {
      posts: [
        { id: 1, title: `Test post` },
      ]
    }
  });

  expect(container).toMatchSnapshot();
});
