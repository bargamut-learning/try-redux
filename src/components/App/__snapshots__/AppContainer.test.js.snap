// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly with mock title 1`] = `
<div>
  <h1>
    Test data
  </h1>
  <form>
    <input />
    <input
      type="submit"
      value="Create post"
    />
  </form>
  <span>
    1
  </span>
  <ul
    class="posts-list"
  >
    <li>
      <button>
        Update
      </button>
      Â 
      Test post
    </li>
  </ul>
  <img />
</div>
`;

exports[`should render App correctly without data 1`] = `
<div>
  <h1>
    Default title
  </h1>
  <form>
    <input />
    <input
      type="submit"
      value="Create post"
    />
  </form>
  <span>
    0
  </span>
  <ul
    class="posts-list"
  >
    <li>
      Loading
      ...
    </li>
  </ul>
  <img />
</div>
`;
